sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/ModelHelper.java

```diff
@@ -504,37 +504,53 @@ public static BlobQueryHeaders transformQueryHeaders(HttpHeaders headers) {
     }
 
     public static void validateConditionsNotPresent(BlobRequestConditions requestConditions,
-        EnumSet<BlobRequestConditionProperty> invalidConditions) {
+        EnumSet<BlobRequestConditionProperty> invalidConditions, String operationName, String parameterName) {
         if (requestConditions == null) {
             return;
         }
-        if (invalidConditions.contains(BlobRequestConditionProperty.LEASE_ID)
-            && requestConditions.getLeaseId() != null) {
-            throw LOGGER.logExceptionAsError(new IllegalArgumentException("'leaseId' is not applicable to this API."));
-        }
-        if (invalidConditions.contains(BlobRequestConditionProperty.TAGS_CONDITIONS)
-            && requestConditions.getTagsConditions() != null) {
-            throw LOGGER.logExceptionAsError(new IllegalArgumentException("'tagsConditions' is not applicable to "
-                + "this API."));
-        }
-        if (invalidConditions.contains(BlobRequestConditionProperty.IF_MODIFIED_SINCE)
-            && requestConditions.getIfModifiedSince() != null) {
-            throw LOGGER.logExceptionAsError(new IllegalArgumentException("'ifModifiedSince' is not applicable to "
-                + "this API."));
-        }
-        if (invalidConditions.contains(BlobRequestConditionProperty.IF_UNMODIFIED_SINCE)
-            && requestConditions.getIfUnmodifiedSince() != null) {
-            throw LOGGER.logExceptionAsError(new IllegalArgumentException("'ifUnmodifiedSince' is not applicable to "
-                + "this API."));
-        }
-        if (invalidConditions.contains(BlobRequestConditionProperty.IF_MATCH)
-            && requestConditions.getIfMatch() != null) {
-            throw LOGGER.logExceptionAsError(new IllegalArgumentException("'ifMatch' is not applicable to this API."));
+        List<String> invalidConditionsFound = new ArrayList<>();
```

Comment:
most people will get it right, i.e. there won't be invalid conditions in API call. eventually they have to - otherwise API won't work. 
Might be good to initialize this list in `if` statements like we did in .net to avoid extra allocations.